### DISCUSSION TITLE
Catch type definition

### DISCUSSION THREAD

**Discussion by bartkoedijk** [ivpusic/react-native-image-crop-picker#1860] [Ideas]
Hi,

First of all I want to thank the creators. maintainers and any other contributors for this library.
I was wondering if we can add a type definition for the catch method while opening the camera and picking an image.

I want to show an error pop up when something went wrong while picking an image.
Currently it will give an ''error'' when the user cancels the operation. I filter it out like this;

```typescript
    .catch((error) => {
         if (error == 'Error: User cancelled image selection') return;
         errorHandler(localeStrings.alertError_ImagePickerOpenLibrary, error);
       }); 
```

This is not ideal since there might be a chance that a maintainer updates/edits this error message.

Thank you for reading.

**Comment by francoangulo** [comment]
Hey @bartkoedijk, I've been making some tests in the catch side, and it seems that when you console.log() the `error`, you get like some strange object but in an array format. But try this `console.log(error.message)`, and you will see that you can access this property like an object, when it seems not to exist when console logging the error "object". The proper way to check if the error is because the user cancel, is to check this condition: `if (error.code === "E_PICKER_CANCELLED")`.
Hope this is useful for you!
